test/__init__.py

```diff
@@ -0,0 +1,250 @@
+            # and not only exists to state that the program is not installed
+            subprocess.check_call([cls._svn, '--version'], stdout=subprocess.DEVNULL)
+        except subprocess.CalledProcessError:
+            raise unittest.SkipTest('`svn` was not found')
+
+        cls.temp_dir = TemporaryDirectory(suffix='.vcstmp')
+
+        # Create the staged subversion repository
+        svnrepo_path = os.path.join(cls.temp_dir.name, 'svnrepo')
+        os.mkdir(svnrepo_path)
+        subprocess.check_call(
+            [
+                'svnadmin',
+                'create',
+                '.',
+            ],
+            cwd=svnrepo_path,
+        )
+
+        svnclone_path = os.path.join(cls.temp_dir.name, 'svnclone')
+        os.mkdir(svnclone_path)
+        shutil.copy(
+            os.path.join(os.path.dirname(os.path.dirname(__file__)), 'LICENSE'),
+            svnclone_path,
+        )
+        for command in (
+            ('checkout', to_file_url(svnrepo_path), '.', '--quiet'),
+            ('add', 'LICENSE', '--quiet'),
+            ('commit', 'LICENSE', '-m', 'Initial commit', '--quiet'),
+        ):
+            subprocess.check_call(
+                [
+                    cls._svn,
+                    *command,
+                ],
+                cwd=svnclone_path,
+            )
+
+        # Create the staged mercurial repository
+        hgrepo_path = os.path.join(cls.temp_dir.name, 'hgrepo')
+        os.mkdir(hgrepo_path)
+        for command in (
+            ('init', '.'),
+            ('branch', '--quiet', 'stable'),
+            ('commit', '-m', 'Initial commit', '-d', cls._commit_date),
+            ('tag', '-d', cls._commit_date, '5.8'),
+        ):
+            subprocess.check_call(
+                [
+                    cls._hg,
+                    *command,
+                ],
+                cwd=hgrepo_path,
+                env=cls._hg_env,
+            )
+
+        # Populate the staged.repos file
+        repos = {
+            'hg/branch': {
+                'type': 'hg',
+                'url': to_file_url(hgrepo_path),
+                'version': 'stable',
+            },
+            'hg/hash': {
+                'type': 'hg',
+                'url': to_file_url(hgrepo_path),
+                'version': '9bd654917508',
+            },
+            'hg/tag': {
+                'type': 'hg',
+                'url': to_file_url(hgrepo_path),
+                'version': '5.8',
+            },
+            'svn/rev': {
+                'type': 'svn',
+                'url': to_file_url(svnrepo_path),
+                'version': '1',
+            },
+        }
+
+        cls.repos_file_path = os.path.join(cls.temp_dir.name, 'staged.repos')
+        with open(cls.repos_file_path, 'w') as f:
+            yaml.safe_dump({'repositories': repos}, f)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.repos_file_path = None
+        cls.temp_dir and cls.temp_dir.cleanup()
```

Comment:
Similarly, a conditional `if` could be added here as well:

```suggestion
        if cls.temp_dir:
            cls.temp_dir.cleanup()
```

Reply:
Same pattern of suggestions as above.